cadence/src/builder.rs

```diff
@@ -47,19 +47,61 @@ impl fmt::Display for MetricType {
 /// implemented but is exposed for documentation purposes and advanced use cases.
 ///
 /// Typical use of Cadence shouldn't require interacting with this type.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub enum MetricValue {
     Signed(i64),
+    PackedSigned(Vec<i64>),
     Unsigned(u64),
+    PackedUnsigned(Vec<u64>),
     Float(f64),
+    PackedFloat(Vec<f64>),
+}
+
+impl MetricValue {
+    fn count(&self) -> usize {
+        match self {
+            Self::PackedSigned(x) => x.len(),
+            Self::PackedUnsigned(x) => x.len(),
+            Self::PackedFloat(x) => x.len(),
+            _ => 1,
+        }
+    }
+}
+
+fn write_value<T>(vals: &[T]) -> String
+where
+    T: fmt::Display,
+{
+    let mut out = String::new();
+
+    for (i, value) in vals.iter().enumerate() {
+        if i > 0 {
+            out.push(':');
+        }
+        out.push_str(format!("{}", value).as_str());
+    }
+
+    out
 }
 
 impl fmt::Display for MetricValue {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match *self {
+        match &*self {
             MetricValue::Signed(v) => v.fmt(f),
+            MetricValue::PackedSigned(v) => {
+                let out = write_value(v);
```

Comment:
I'm not super familiar with rust string formatters. Let me know if there's a better way of doing this!

Reply:
I think the `write_value` method should accept the `Formatter` as an argument, writing each value with either `Formatter::write_str` or `T::fmt(f)`. That way we aren't allocating a new string each iteration of the loop with `format!()`

Reply:
Done in https://github.com/56quarters/cadence/pull/169/commits/d505641bd660006f49bfcd687ceb0898c411e9df!